Public/Sdk/Public/Managed/Tools/Csc/csc.dsc

```diff
@@ -76,7 +76,10 @@ export function compile(inputArgs: Arguments) : Result {
 
         Cmd.option("/langversion:", args.languageVersion),
 
-        Cmd.option("/define:",       args.defines ? args.defines.join(";")    : undefined),
+        // TODO: uncoment the following line and delete the line after it once a new LKG is published
+        // Cmd.option("/define:",       args.defines ? args.defines.join(";") : undefined), 
+        ...addIf((args.defines || []).length > 0, Cmd.rawArgument(`/define:"${args.defines.join(';')}"`)),

```

Comment:
>d.rawArgume [](start = 52, length = 11)

I think the fix would be better to have a new enum for argument escaping that matches Unix'es Process launch encodings....  and have that algorithm be the default on the mac.

Okay for now but just sharing if we hit more of these in the future.

Reply:
Indeed.